### PR TITLE
Initial deb and rpm packaging

### PR DESCRIPTION
**Pull request by jchengsfx**
**Description:**
- Uses https://github.com/jordansissel/fpm
- Adds `deb-package` and `rpm-package` targets to Makefile
- Builds simple deb and rpm packages that installs the otelcol
binary to `/usr/bin/otelcol` and `examples/otel-local-config.yaml`
to `/etc/otel-collector/config.yaml`.
- Adds deb and rpm packages to circleci workspace for github release
- TODO: create service scripts

**Link to tracking Issue:** <Issue number if applicable>

**Testing:**
- Added simple installation test after package build

**Documentation:**
- Added `packaging/deb/README.md` and `packaging/rpm/README.md`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -334,3 +350,35 @@ jobs:
...
+      package_type:
+        type: enum
+        enum: ["deb", "rpm"]
+    steps:
+      - checkout
+      - attach_to_workspace
+      - run:
+          name: Install fpm and dependencies
+          command: |
+            sudo apt-get update
```
Comment (flands) [inline .circleci/config.yml:366]: should an `upgrade` happen here as well?

File: packaging/deb/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM debian:9
+
+RUN apt-get update && \
```
Comment (flands) [inline packaging/deb/Dockerfile:3]: `upgrade` here as well?

**Comment by jchengsfx** [comment]
Need to make similar changes according to review in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/405.

File: .circleci/config.yml (hunk 1 above)
Comment (bogdandrutu) [reply .circleci/config.yml:366]: `upgrade` upgrades other deps, update just refreshes the repositories.

**Pull request merged**